tests/testthat/test-test-error.R

```diff
@@ -0,0 +1,41 @@
+test_that("API endoint errors, status API errors too", {
+  library("magrittr")
+  webmockr::enable()
+
+  webmockr::stub_request("get", paste0(base_url(), "countries?limit=0&page=1")) %>%
+    webmockr::to_return(status = 404)
+
+  webmockr::stub_request("get", status_url()) %>%
+    webmockr::to_return(body = '{"meta":{"name":"openaq-api","license":"CC BY 4.0","website":"https://docs.openaq.org/"},"results":{"healthStatus":"red"}}')
+
+  expect_error(aq_countries(), "uh oh")
+
+  webmockr::disable()
+})
+
+test_that("API errors twice, status not", {
+  library("magrittr")
+  # when not all is well
+  webmockr::enable()
+
+  webmockr::stub_request("get", paste0(base_url(), "countries?limit=0&page=1")) %>%
+    webmockr::to_return(status = 404, body = "retry!", times = 2) %>%
+    webmockr::to_return(status = 200,
+                        headers = list(date = "Thu, 09 Apr 2020 14:20:43 GMT"),
+                        body = '{"meta":{"name":"openaq-api","license":"CC BY 4.0","website":"https://docs.openaq.org/","page":1,"limit":100,"found":98}}') # here I want to use a real response and cassette
```

Comment:
E.g. how do I use a cassette here instead of having to copy-paste a body from the API? I only want to write by hand for the fake things (errors)

Reply:
I've thought about this some. I don't think there's any easy way to do this. perhaps though you can use vcr above this block to cache the request/response. then supply the response you want to `body` here from the vcr block